src/rulesets/__tests__/reader.jest.test.ts

```diff
@@ -353,40 +337,21 @@ describe('Rulesets reader', () => {
     ).toHaveLength(0);
   });
 
-  it('should limit the scope of formats to a ruleset', () => {
-    return expect(readRuleset(myOpenAPIRuleset)).resolves.toEqual(
-      expect.objectContaining({
-        rules: {
-          ...Object.entries(oasRuleset.rules).reduce<Dictionary<unknown>>((rules, [name, rule]) => {
-            rules[name] = expect.objectContaining({
-              formats: ['oas2', 'oas3'],
-            });
+  it('should limit the scope of formats to a ruleset', async () => {
+    const rules = (await readRuleset(myOpenAPIRuleset)).rules;
 
-            return rules;
-          }, {}),
+    expect(Object.keys(rules)).toHaveLength(4);
 
-          ...Object.entries(oas2Ruleset.rules).reduce<Dictionary<unknown>>((rules, [name, rule]) => {
-            rules[name] = expect.objectContaining({
-              formats: ['oas2'],
-            });
+    expect(rules['my-valid-rule'].formats).toBeUndefined();
 
-            return rules;
-          }, {}),
+    expect(rules['generic-valid-rule'].formats).toBeInstanceOf(Array);
```

Comment:
```suggestion
```

Reply:
(Unneeded since we match against an array of oas2 and oas3)

Reply:
Fixed